### PR TITLE
Add fonts Noto Emoji

### PR DESCRIPTION
**Pull request by puripuri2100**
Add [`satysfi-fonts-noto-emoji.1.0.0`](https://github.com/puripuri2100/SATySFi-fonts-noto-emoji).

"SATySFi font package for Noto Emoji fonts."

### CODE REVIEW

File: packages/satysfi-fonts-noto-emoji/satysfi-fonts-noto-emoji.1.0.0/opam [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+maintainer: "Naoki Kaneko <puripuri2100@gmail.com>"
+authors: "Naoki Kaneko <puripuri2100@gmail.com>"
+license: "MIT"
+homepage: "https://github.com/puripuri2100/SATySFi-fonts-noto-emoji"
+bug-reports: "https://github.com/puripuri2100/SATySFi-fonts-noto-emoji/issues"
+dev-repo: "git+https://github.com/puripuri2100/SATySFi-fonts-noto-emoji.git"
+
+
+extra-source "NotoEmoji-unhinted.zip" {
+  archive: "https://noto-website-2.storage.googleapis.com/pkgs/NotoEmoji-unhinted.zip"
```
Comment (na4zagin3) [inline packages/satysfi-fonts-noto-emoji/satysfi-fonts-noto-emoji.1.0.0/opam]: If you want to use `extra-source`, can you add `checksum` to each `extra-source`?

File: packages/satysfi-fonts-noto-emoji/satysfi-fonts-noto-emoji.1.0.0/opam [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+maintainer: "Naoki Kaneko <puripuri2100@gmail.com>"
+authors: "Naoki Kaneko <puripuri2100@gmail.com>"
+license: "MIT"
+homepage: "https://github.com/puripuri2100/SATySFi-fonts-noto-emoji"
+bug-reports: "https://github.com/puripuri2100/SATySFi-fonts-noto-emoji/issues"
+dev-repo: "git+https://github.com/puripuri2100/SATySFi-fonts-noto-emoji.git"
+
+
+extra-source "NotoEmoji-unhinted.zip" {
```
Comment (na4zagin3) [inline packages/satysfi-fonts-noto-emoji/satysfi-fonts-noto-emoji.1.0.0/opam]: I would get files from <https://github.com/googlefonts/noto-emoji/tree/master/fonts>, using [`git submodule`](https://qiita.com/sotarok/items/0d525e568a6088f6f6bb).

File: .travis.yml [hunk 3]
```diff
@@ -22,6 +22,7 @@ env:
       satysfi-fonts-computer-modern-unicode-doc
       satysfi-fonts-dejavu
       satysfi-fonts-dejavu-doc
+      satysfi-fonts-noto-emoji
```
Comment (na4zagin3) [inline .travis.yml]: Can you create `-doc` library with font specimen if you have time?

File: packages/satysfi-fonts-noto-emoji/satysfi-fonts-noto-emoji.1.0.0/opam [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+synopsis: "SATySFi Font Package for Noto Emoji fonts"
+description: """
+SATySFi font package for Noto Emoji fonts.
+
+This package installs fonts from https://www.google.com/get/noto/.
+"""
+
+maintainer: "Naoki Kaneko <puripuri2100@gmail.com>"
+authors: "Naoki Kaneko <puripuri2100@gmail.com>"
+license: "MIT"
```
Comment (na4zagin3) [inline packages/satysfi-fonts-noto-emoji/satysfi-fonts-noto-emoji.1.0.0/opam]: The Noto fonts’ license is [`OFL-1.1`](https://spdx.org/licenses/OFL-1.1.html). See https://github.com/googlefonts/noto-emoji#license

File: packages/satysfi-fonts-noto-emoji/satysfi-fonts-noto-emoji.1.0.0/opam [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+]
+install: [
+  ["satyrographos" "opam" "install"
+   "-name" "fonts-noto-emoji"
+   "-prefix" "%{prefix}%"
+   "-script" "%{build}%/Satyristes"]
+]
+url: {
+  src: "https://github.com/puripuri2100/SATySFi-fonts-noto-emoji/archive/1.0.0.tar.gz"
+}
```
Comment (na4zagin3) [inline packages/satysfi-fonts-noto-emoji/satysfi-fonts-noto-emoji.1.0.0/opam]: Nitpick: Generally speaking, every text file is supposed to have a line break at the end. See <https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline>.

**Comment by na4zagin3** [comment]
@puripuri2100 Can you change the version for `1.05+uh` since the version embedded in NotoEmoji is `1.05 uh`, although I don’t understand what `uh` stands for?

File: packages/satysfi-fonts-noto-emoji-doc/satysfi-fonts-noto-emoji-doc.1.0.0/opam [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+authors: "Naoki Kaneko <puripuri2100@gmail.com>"
+license: "CC0-1.0"
+homepage: "https://github.com/puripuri2100/SATySFi-fonts-noto-emoji"
+bug-reports: "https://github.com/puripuri2100/SATySFi-fonts-noto-emoji/issues"
+dev-repo: "git+https://github.com/puripuri2100/SATySFi-fonts-noto-emoji.git"
+depends: [
+  "satysfi" {>= "0.0.3" & < "0.0.5"}
+  "satyrographos" {>= "0.0.2" & < "0.0.3"}
+  "satysfi-base" {>= "1.2.0"}
+  "satysfi-fonts-noto-emoji" {= "1.0.1"}
```
Comment (na4zagin3) [inline packages/satysfi-fonts-noto-emoji-doc/satysfi-fonts-noto-emoji-doc.1.0.0/opam]: Version restriction is wrong.
```
The following dependencies couldn't be met:

          - satysfi-fonts-noto-emoji-doc → satysfi-fonts-noto-emoji >= 1.0.1

              no matching version
```
https://travis-ci.com/na4zagin3/satyrographos-repo/jobs/295771330

File: packages/satysfi-fonts-noto-emoji/satysfi-fonts-noto-emoji.1.05+uh/opam [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+synopsis: "SATySFi Font Package for Noto Emoji fonts"
+description: """
+SATySFi font package for Noto Emoji fonts.
+
+This package installs fonts from https://www.google.com/get/noto/.
+"""
+
+maintainer: "Naoki Kaneko <puripuri2100@gmail.com>"
+authors: "Naoki Kaneko <puripuri2100@gmail.com>"
+license: "OFL1-1"
```
Comment (na4zagin3) [inline packages/satysfi-fonts-noto-emoji/satysfi-fonts-noto-emoji.1.05+uh/opam]: Can you fix this typo? SPDX License Identifier of OFL v1.1 is [`OFL-1.1`](https://spdx.org/licenses/OFL-1.1.html) but not `OFL1-1`.

File: packages/satysfi-fonts-noto-emoji-doc/satysfi-fonts-noto-emoji-doc.1.05+uh/opam [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+authors: "Naoki Kaneko <puripuri2100@gmail.com>"
+license: "CC0-1.0"
+homepage: "https://github.com/puripuri2100/SATySFi-fonts-noto-emoji"
+bug-reports: "https://github.com/puripuri2100/SATySFi-fonts-noto-emoji/issues"
+dev-repo: "git+https://github.com/puripuri2100/SATySFi-fonts-noto-emoji.git"
+depends: [
+  "satysfi" {>= "0.0.3" & < "0.0.5"}
+  "satyrographos" {>= "0.0.2" & < "0.0.3"}
+  "satysfi-base" {>= "1.2.0"}
+  "satysfi-fonts-noto-emoji" {>= "1.05+uh"}
```
Comment (na4zagin3) [inline packages/satysfi-fonts-noto-emoji-doc/satysfi-fonts-noto-emoji-doc.1.05+uh/opam]: Can you use `=` for the corresponding library (not for `satysfi-base`)? Library version and document version need to be exactly the same. 

```suggestion
  "satysfi-fonts-noto-emoji" {= "1.05+uh"}
```

**Comment by puripuri2100** [comment]
I am sorry for the inconvenience.🙇

**Comment by na4zagin3** [comment]
It’s not inconvenience at all! But GitHub missing the CI job completion may be... 

Anyway, the CI passed (https://travis-ci.com/na4zagin3/satyrographos-repo/builds/152485104).

I’m merging this. Thank you!

File: packages/satysfi-fonts-noto-emoji/satysfi-fonts-noto-emoji.1.0.0/opam (hunk 2 above)
Comment (na4zagin3) [reply packages/satysfi-fonts-noto-emoji/satysfi-fonts-noto-emoji.1.0.0/opam]: This is a issue shared with the other Noto font packages. I’m resolving this for now.

**Comment by zeptometer** [comment]
> @puripuri2100 Thank you for sending PR! Could you please address the points I pointed out?
> 
> You may probably use @zeptometer’s https://github.com/zeptometer/create-satysfi-fonts-package which apparently generated the other `satysfi-fonts-noto-*` libraries.
> 
> @zeptometer, can you confirm that?

Sorry for late reply to this. Currently, I don't recommend to use [create-satysfi-fonts-package](https://github.com/zeptometer/create-satysfi-fonts-package) as it totally lacks documents and some values are hardcoded (which might be wrong in the future). I hope I can release it in the near future though :)

BTW, thank you for creating this font library @puripuri2100 !

**Pull request merged**